File: Largest prime factor.java
class Solution{
    static long largestPrimeFactor(int N) {
        // code here
        
        for (int i =2; i*i<=N ; i++){
            if (N%i==0){
                N = N/i;
                i--;
            }
        }
       return  (long)N;
    }
}

File: Largest prime factor.cpp
 long long int largestPrimeFactor(int N){
        // code here
       // by factorial approach
       
       long long fact =2;  //minimum prime is 2
       
       while(N){
           
           if(N%fact == 0)  //hence N is not prime number
          { 
           N = N/fact;
           
           if(N==1) break;
          }
          else {
              fact++;
          }
       }
       return fact;
    }
